File: app/src/main/java/com/example/cleanarchitecture_toyproject/data/user/UserCacheSource.java
package com.example.cleanarchitecture_toyproject.data.user;

import com.example.cleanarchitecture_toyproject.data.entity.UserEntity;
import com.example.cleanarchitecture_toyproject.viewmodel.model.UserModel;
import io.reactivex.Observable;

import java.util.List;

public interface UserCacheSource {

    //Insert Favorite User
    //Observable<UserEntity> setUsers(UserModel user);
    public void setUsers(UserEntity userEntity);

    //Observable<UserModel> getUserList();
    UserEntity getUserList(int id);

    //select Favorite User
    Observable<List<UserEntity>> selectUsers();

    //delete Favorite User
    public void deleteUsers(UserEntity userEntity);
}

File: app/src/main/java/com/example/cleanarchitecture_toyproject/data/user/UserCacheSource.kt
package com.example.cleanarchitecture_toyproject.data.user

import com.example.cleanarchitecture_toyproject.data.entity.UserEntity
import com.example.cleanarchitecture_toyproject.viewmodel.model.UserModel
import io.reactivex.Observable

interface UserCacheSource {

    //Insert Favorite User
    //Observable<UserEntity> setUsers(UserModel user);
    fun setUsers(userEntity: UserEntity)

    //Observable<UserModel> getUserList();
    fun getUserList(id: Int): UserEntity

    //select Favorite User
    fun selectUsers(): Observable<List<UserEntity>>

    //delete Favorite User
    fun deleteUsers(userEntity: UserEntity)
}
